go/processor/floria/processor.go

```diff
@@ -49,15 +49,43 @@ func init() {
 // - Consume 10% of the remaining gas
 func newFloriaProcessor(interpreter tosca.Interpreter) tosca.Processor {
 	return &Processor{
-		Interpreter:   interpreter,
-		EthCompatible: false,
+		Interpreter: interpreter,
+		Config: Config{
+			EthCompatible: false,
+			StateContracts: map[tosca.Address]StateContract{
+				StateContractAddress(): StateContractSonic{},
+			},
+		},
 	}
 }
 
 // Processor implements the tosca.Processor interface for the Floria processor.
 type Processor struct {
-	Interpreter   tosca.Interpreter
+	Interpreter tosca.Interpreter
+	Config      Config
+}
+
+type StateContract interface {
+	Run(state tosca.WorldState, sender tosca.Address, receiver tosca.Address, input []byte, gas tosca.Gas) tosca.CallResult
+}
+
+// Config holds configuration options for the Floria processor.
+type Config struct {
+	// EthCompatible flag indicates whether the processor is compatible with Ethereum.
+	// The default is compatibility with Sonic/Fantom.
 	EthCompatible bool
+
+	// StateContracts allows to add special contracts that can not be implemented as
```

Comment:
"that can not" does not work very well.

```suggestion
	// StateContracts binds addresses to special contracts not deployed in the blockchain. 
```